"py uninstall --purge" crashes with INTERNAL ERROR

**Install source and version**
- [ ] Installed from the Windows Store
- [ ] Installed with the MSIX from python\.org
- [ ] Installed with the MSI from python\.org
- [x] Installed with `winget install 9NQ7512CXL7T`

Version: 26.0

**Describe the bug**
Running `py uninstall --purge` crashes with an internal error:
```
[ERROR] INTERNAL ERROR: TypeError: 'NoneType' object is not callable
```

The command starts normally and removes runtimes and cache directories, but fails during the final stage while purging shortcuts.

**To Reproduce**
1. Install Python Install Manager (MSIX from python.org)
2. Install at least one runtime with `py install` (tried with 3.10 and latest 3.14)
3. Run:
   ```
   py uninstall --purge
   ```
4. Confirm with `Y` or `Enter`
5. Observe crash after "Purging all shortcuts"

**Expected behavior**
`py uninstall --purge` should complete successfully without throwing an internal exception.

**Additional context**
```logs
>> Writing logs to C:\Users\Lusiiky\AppData\Local\Temp\python_uninstall_20260322153836_7300.log
>> Python installation manager 26.0
Copyright (c) Python Software Foundation. All Rights Reserved.

BEGIN uninstall_command.execute: []
Reading installs from C:\Users\Lusiiky\AppData\Local\Python
Found 1 install
Calculating default install
Filtering installs by tag = 3
tag '3' matched 1 install
- 1 core install(s) by prefix
default_platform '-64' matched 1 install
Default install will be pythoncore-3.10-64
>  Purging Python 3.10.11 from C:\Users\Lusiiky\AppData\Local\Python\pythoncore-3.10-64
Atomically removing these files first: python.exe, python3.dll, python310.dll, pythonw.exe, vcruntime140.dll, vcruntime140_1.dll, __install__.json
Files successfully removed
>  Purging saved downloads from C:\Users\Lusiiky\AppData\Local\Python\_cache
Current PATH contains C:\Users\Lusiiky\AppData\Local\Microsoft\WindowsApps;D:\Users\Lusiiky\AppData\Local\Programs\Microsoft VS Code\bin;C:\Users\Lusiiky\AppData\Local\Muse Hub\lib;C:\Users\Lusiiky\.lmstudio\bin;C:\Users\Lusiiky\AppData\Local\JetBrains\Toolbox\scripts;%USERPROFILE%\AppData\Local\Microsoft\WindowsApps;C:\Users\Lusiiky\AppData\Local\Microsoft\WinGet\Packages\derrod.legendary_Microsoft.Winget.Source_8wekyb3d8bbwe;;C:\Users\Lusiiky\AppData\Local\Programs\ExifTool;C:\Users\Lusiiky\AppData\Roaming\npm;C:\Users\Lusiiky\AppData\Local\PowerToys\DSCModules\
>  Purging global commands from C:\Users\Lusiiky\AppData\Local\Python\bin
Purging C:\Users\Lusiiky\AppData\Local\Python\bin\pip.exe
Purging C:\Users\Lusiiky\AppData\Local\Python\bin\pip.exe.__script__.py
Purging C:\Users\Lusiiky\AppData\Local\Python\bin\pip.exe.__target__
Purging C:\Users\Lusiiky\AppData\Local\Python\bin\pip3.9.exe
Purging C:\Users\Lusiiky\AppData\Local\Python\bin\pip3.9.exe.__script__.py
Purging C:\Users\Lusiiky\AppData\Local\Python\bin\pip3.9.exe.__target__
Purging C:\Users\Lusiiky\AppData\Local\Python\bin\pip3.exe
Purging C:\Users\Lusiiky\AppData\Local\Python\bin\pip3.exe.__script__.py
Purging C:\Users\Lusiiky\AppData\Local\Python\bin\pip3.exe.__target__
Purging C:\Users\Lusiiky\AppData\Local\Python\bin\python.exe
Purging C:\Users\Lusiiky\AppData\Local\Python\bin\python.exe.__target__
Purging C:\Users\Lusiiky\AppData\Local\Python\bin\python3.10.exe
Purging C:\Users\Lusiiky\AppData\Local\Python\bin\python3.10.exe.__target__
Purging C:\Users\Lusiiky\AppData\Local\Python\bin\python3.exe
Purging C:\Users\Lusiiky\AppData\Local\Python\bin\python3.exe.__target__
Purging C:\Users\Lusiiky\AppData\Local\Python\bin\pythonw.exe
Purging C:\Users\Lusiiky\AppData\Local\Python\bin\pythonw.exe.__target__
Purging C:\Users\Lusiiky\AppData\Local\Python\bin\pythonw3.10.exe
Purging C:\Users\Lusiiky\AppData\Local\Python\bin\pythonw3.10.exe.__target__
Purging C:\Users\Lusiiky\AppData\Local\Python\bin\pythonw3.exe
Purging C:\Users\Lusiiky\AppData\Local\Python\bin\pythonw3.exe.__target__
>  Purging all shortcuts
Cleaning up registry entries
Removing HKEY_CURRENT_USER\Software\Python\PythonCore\3.10
Cleaning up Start menu shortcuts
Removing C:\Users\Lusiiky\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\Python\Python 3.10\Python 3.10.lnk
Removing C:\Users\Lusiiky\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\Python\Python 3.10\Python 3.10 Online Documentation.lnk
Removing C:\Users\Lusiiky\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\Python\Python 3.10
Removing C:\Users\Lusiiky\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\Python
Removing ARP registration for pymanager-pythoncore-3.10-64
!! INTERNAL ERROR: TypeError: 'NoneType' object is not callable
TRACEBACK:
Traceback (most recent call last):
  File "manage\__init__.py", line 50, in main
  File "manage\commands.py", line 900, in execute
  File "manage\uninstall_command.py", line 109, in execute
TypeError: 'NoneType' object is not callable
```

--- a/src/manage/pep514utils.py
+++ b/src/manage/pep514utils.py
@@ -224,18 +224,20 @@ def _is_tag_managed(company_key, tag_name, *, creating=False, allow_warn=True):
 def _split_root(root_name):
     if not root_name:
         LOGGER.verbose("Skipping registry shortcuts as PEP 514 registry root is not set.")
-        return
+        return None, None
     hive_name, _, name = root_name.partition("\\")
     try:
         hive = getattr(winreg, hive_name.upper())
     except AttributeError:
-        LOGGER.verbose("Skipping registry shortcuts as %s\\%s is not a valid key", root_name)
-        return
+        LOGGER.verbose("Skipping registry shortcuts as %s\\%s is not a valid key", root_name, hive_name)
+        return None, None
     return hive, name
 
 
 def update_registry(root_name, install, data, warn_for=[]):
     hive, name = _split_root(root_name)
+    if not hive or not name:
+        return
     with winreg.CreateKey(hive, name) as root:
         allow_warn = install_matches_any(install, warn_for)
         if _is_tag_managed(root, data["Key"], creating=True, allow_warn=allow_warn):
@@ -258,6 +260,8 @@ def update_registry(root_name, install, data, warn_for=[]):
 def cleanup_registry(root_name, keep, warn_for=[]):
     LOGGER.debug("Cleaning up registry entries")
     hive, name = _split_root(root_name)
+    if not hive or not name:
+        return
     with _reg_open(hive, name, writable=True) as root:
         for company_name in list(_iter_keys(root)):
             any_left = False

--- a/src/manage/uninstall_command.py
+++ b/src/manage/uninstall_command.py
@@ -106,7 +106,8 @@ def execute(cmd):
         _do_purge_global_dir(cmd.global_dir, warn_msg.format("global commands"))
         LOGGER.info("Purging all shortcuts")
         for _, cleanup in SHORTCUT_HANDLERS.values():
-            cleanup(cmd, [])
+            if cleanup:
+                cleanup(cmd, [])
         LOGGER.debug("END uninstall_command.execute")
         return
 

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -148,10 +148,16 @@ def localserver():
                 p.wait(5)
 
 
+REG_TEST_ROOT = r"Software\Python\PyManagerTesting"
+
+
 class FakeConfig:
     def __init__(self, global_dir, installs=[]):
-        self.root = global_dir.parent if global_dir else None
         self.global_dir = global_dir
+        self.root = global_dir.parent if global_dir else None
+        self.download_dir = self.root / "_cache" if self.root else None
+        self.start_folder = self.root / "_start" if self.root else None
+        self.pep514_root = REG_TEST_ROOT
         self.confirm = False
         self.installs = list(installs)
         self.shebang_can_run_anything = True
@@ -175,13 +181,12 @@ class FakeConfig:
         return [i for i in self.installs
                 if (not tag or i["tag"] == tag) and (not company or i["company"] == company)][0]
 
+    def ask_yn(self, question):
+        return False if self.confirm else True
 
 @pytest.fixture
 def fake_config(tmp_path):
     return FakeConfig(tmp_path / "bin")
-
-
-REG_TEST_ROOT = r"Software\Python\PyManagerTesting"
 
 
 class RegistryFixture:

--- a/tests/test_uninstall_command.py
+++ b/tests/test_uninstall_command.py
@@ -17,3 +17,11 @@ def test_purge_global_dir(monkeypatch, registry, tmp_path):
     assert registry.getvalueandkind("", "Path") == (
         rf"C:\A;{tmp_path}\X;C:\B;%PTH%;C:\%D%\E", winreg.REG_SZ)
     assert not list(tmp_path.iterdir())
+
+
+def test_null_purge(fake_config):
+    cmd = fake_config
+    cmd.args = ["--purge"]
+    cmd.confirm = False
+    cmd.purge = True
+    UC.execute(cmd)

